### ISSUE TITLE
pycsope dose not work on MAC OS X

### ISSUE THREAD

**Issue by hujianxin** [syl20bnr/spacemacs#3256]
Emacs version: 24.5.1
OS version: OS X 10.10 & Ubuntu 14.04
On ubuntu platform, pycscope works well with `SPC m g i` `SPC m g r` and so on.
On OS X platform, when I entered 'SPC m g r', emacs show nothing, but, this action worked well on ubuntu. p.s. The Python project is the same.

**Comment by robbyoconnor** [comment]
@hujianxin -- `SPC hds`

**Comment by hujianxin** [comment]
#### System Info
- OS: darwin
- Emacs: 24.5.1
- Spacemacs: 0.104.2
- Spacemacs branch: master (rev. 074f425)
- Distribution: spacemacs
- Layers:
```elisp
(osx auto-completion better-defaults git
     (shell :variables shell-default-height 30 shell-default-position 'bottom)
     spell-checking syntax-checking version-control ranger dockerfile react django emoji asciidoc c-c++ markdown org emacs-lisp ess html ipython-notebook java javascript latex python scala shell-scripts sql yaml pandoc evil-commentary evil-snipe unimpaired vim-empty-lines vim-powerline vinegar eyebrowse perspectives chinese colors cscope deft ibuffer search-engine smex themes-megapack)
```

**Comment by robbyoconnor** [comment]
Things to try: 

- Update packages using the `[Update]` button the home buffer
- See if `develop` works -- odds are it won't be useful to do this but hopefully the first fixes shit.

**Comment by hujianxin** [comment]
I have did following what you said.
+ `cd ~/.emacs.d`
+ `git checkout -b develop origin/develop`
And then, the pycscope did not work as well. :disappointed:

**Comment by hujianxin** [comment]
I know little emacs lisp and I'm going to learn it. But now , I cannot solve this bug(if it is) :sob:

**Comment by robbyoconnor** [comment]
I suspect this is a `PATH` issue

**Comment by hujianxin** [comment]
I can't image what cause this issue. I use oh-my-zsh, and in shell environment, pyenv works well.
And, in emacs, pyenv works, because, when I select diferent env by `SPC m v s`, the complete messages are diferent. Which makes completing messages difereent is that I install diferent packages under diferent env.

**Comment by hujianxin** [comment]
@robbyoconnor What os are you using?

**Comment by robbyoconnor** [comment]
#### System Info
- OS: gnu/linux
- Emacs: 24.4.1
- Spacemacs: 0.105.0
- Spacemacs branch: develop (rev. b92eb41)
- Distribution: spacemacs
- Layers:
```elisp
((auto-completion :variables auto-completion-return-key-behavior 'complete auto-completion-tab-key-behavior 'cycle auto-completion-complete-with-key-sequence nil auto-completion-enable-help-tooltip t auto-completion-enable-sort-by-usage t auto-completion-show-snippets-in-popup t)
 better-defaults emacs-lisp
 (git :variables git-magit-status-fullscreen t)
 github gnus version-control markdown syntax-checking
 (latex :variables latex-enable-auto-fill t)
 (colors :variables colors-enable-rainbow-identifiers t colors-enable-nyan-cat-progress-bar t)
 dockerfile ansible puppet evil-commentary
 (evil-snipe :variables evil-snipe-enable-alternate-f-and-t-behaviors t)
 fasd finance floobits xkcd autohotkey
 (c-c++ :variables c-c++-enable-clang-support t)
 (clojure :variables clojure-enable-fancify-symbols t)
 extra-langs go
 (haskell :variables haskell-enable-ghci-ng-support t haskell-enable-shm-support t haskell-enable-hindent-style "andrew-gibiansky")
 html java javascript
 (python :variables python-enable-yapf-format-on-save t)
 racket
 (ruby :variables ruby-version-manager `rvm)
 ruby-on-rails rust scala shell-scripts restclient themes-megapack tmux vim-empty-lines spotify pandoc vagrant
 (ibuffer :variables ibuffer-group-buffers-by 'projects)
 semantic deft
 (shell :variables shell-default-shell 'eshell shell-default-position 'bottom shell-default-height 30 shell-default-term-shell "/bin/zsh")
 typescript
 (eyebrowse :variables eyebrowse-display-help t)
 (perspectives :variables perspective-display-help t)
 erc chrome d extra-langs smex emoji gtags prodigy evernote
 (org :variables org-enable-github-support t)
 search-engine evil-little-word yaml sql nim ipython-notebook lua scheme purescript sml common-lisp ranger wakatime dash spell-checking jabber deft cscope vinegar unimpaired rcirc games react)
```

**Comment by hujianxin** [comment]
Oh, thanks. I expect someone using mac can tell me whether he met the same problem. If not, I can confirm that's a PATH issue.

**Comment by CarlQLange** [comment]
I can have a look, any details I should know? I don't use pyscope

**Comment by hujianxin** [comment]
Hi, @CarlQLange . You can do it as following:
+ `brew install cscope`
+ `sudo pip install pycscope`
+ Startup Emacs and dir to a python project(git project).
+ Use `SPC m g i` to generate cscope index.
+ Use `SPC m g r` to find all of the <word> which under the cursor in the project.
+ If there's nothing finded, but there's other place that <word> appears.It says your pycscope dose not work as well. If path that all of  <word> appears pops up, It says that your pycscope works well.

My english is broken, can you understand?

**Comment by cpaulik** [comment]
@hujianxin Which functions do the `SPC m g i` and `SPC m g r` keybindings call? They are not defined in Spacemacs are they?

**Comment by hujianxin** [comment]
@cpaulik You shoud add cscope support to .spacemacs file. :smile:

**Comment by bmag** [comment]
@hujianxin I'm using Ubuntu, but I think I can help.
When `, g r` doesn't find any references, it will show an empty helm buffer, which I assume is what you meant when you wrote "nothing". On the other hand, if it's a path issue then `, g r` should print an error message: "Searching for program: no such file or directory, cscope".

`, g i` runs pycscope, which creates a cscope index file (`cscope.out`). `, g r` runs `cscope` to query the index file. Did you check that pycscope and cscope work correctly from the command line? To check, open a terminal and do the following:
- 1. go to your project directory and run `pycscope -R -f cscope.out`.
- 2. to find all references of <symbol>, run `cscope -d -f cscope.out -L -0 <symbol>` - this is roughly what `, g r` runs. The `-d` is important so cscope doesn't regenerate cscope.out.
If successful, cscope will print all references of <symbol> in your project.

**Comment by hujianxin** [comment]
@bmag Hi, thank you very much. Thanks to your guidance, I finded that cscope works well in other directory, but not here. And in command line, I got a error message "cscope: cannot read trailer offset from file cscope.out"，so I googled it. Finally, I found the reason that Cscope cannot seem to handle paths with spaces. So i changed path name to one without spaces, and solved this problem.
Thanks again.
@robbyoconnor The problem was solved, thanks too.
I'll close this issue.
